packages/swap/src/routes/v2/swap.ts

```diff
@@ -132,8 +133,12 @@ router.get(
       getEgressStatusFields(swapRequest, failedSwap, 'SWAP', swapEgressTrackerTxRef),
       getEgressStatusFields(swapRequest, failedSwap, 'REFUND', refundEgressTrackerTxRef),
       internalDestAsset &&
-        estimateSwapDuration({ srcAsset: internalSrcAsset, destAsset: internalDestAsset }),
-      getRequiredBlockConfirmations(internalSrcAsset),
+        estimateSwapDuration({
+          srcAsset: internalSrcAsset,
+          destAsset: internalDestAsset,
+          isExternal,
+        }),
+      isExternal ? getRequiredBlockConfirmations(internalSrcAsset) : Promise.resolve(undefined),
```

Comment:
you don't have to pass only promises here
```suggestion
      isExternal ? getRequiredBlockConfirmations(internalSrcAsset) : undefined,
```

Reply:
Changed